### DISCUSSION TITLE
Apache Breeze - Problem installing Python packages with init.sh script

### DISCUSSION THREAD

**Discussion by hrvoje-intuita** [apache/airflow#29356] [Q&A] [answered]
### Apache Airflow version

main (development)

### What happened

Hi,

I'm using Apache Breeze on WSL2 Windows 11 machine. 
Breeze is installed without problems and runs images fine. When running with empty init.sh all runs well and Airflow starts ok for this command:
```
breeze start-airflow --python 3.10  --load-example-dags --backend postgres --postgres-version 14 --use-airflow-version 2.5.0 │
 --airflow-constraints-reference "https://raw.githubusercontent.com/apache/airflow/constraints-2.5.0/constraints-3.10.txt" |
 --airflow-extras [pandas,amazon,postgres,sftp]
```
It's worth noting that pip report this during airflow dependencies installation:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
types-paramiko 2.12.0.3 requires cryptography>=37.0.0, but you have cryptography 36.0.2 which is incompatible.
sphinx-rtd-theme 1.1.1 requires docutils<0.18, but you have docutils 0.19 which is incompatible.
pyopenssl 22.1.0 requires cryptography<39,>=38.0.0, but you have cryptography 36.0.2 which is incompatible.
```
Regardless of that airflow server starts ok. 

**Problem description**
I'm trying to install Python packages used in DAGs with file
`files/airflow-breeze-config/init.sh` 
as advised [here](https://github.com/apache/airflow/blob/08ec820b167ab3b2489ad8e4fb900d9390cfea49/BREEZE.rst#customizing-your-environment). 

My init.sh:
```
pip install --upgrade pip
pip install -r /opt/airflow/dags/requirements.txt
```

First command runs fine, second one gives error:

> Collecting airflow==0.6
>   Downloading airflow-0.6.tar.gz (1.2 kB)
>   Preparing metadata (setup.py) ... error
>   error: subprocess-exited-with-error
> 
>   × python setup.py egg_info did not run successfully.
>   │ exit code: 1
>   ╰─> [14 lines of output]
>       running egg_info
>       creating /tmp/pip-pip-egg-info-6vzwl974/airflow.egg-info
>       writing /tmp/pip-pip-egg-info-6vzwl974/airflow.egg-info/PKG-INFO
>       writing dependency_links to /tmp/pip-pip-egg-info-6vzwl974/airflow.egg-info/dependency_links.txt
>       writing top-level names to /tmp/pip-pip-egg-info-6vzwl974/airflow.egg-info/top_level.txt

My requirements.txt looks like this:
```
pyOpenSSL==22.0.0
boto3==1.26.50
botocore==1.23.34
jsonpath_ng==1.5.3
numpy==1.24.1
pandas==1.5.2
pendulum==2.1.2
psycopg2==2.9.5
psycopg2_binary==2.9.5
python_dateutil==2.8.2
redshift_connector==2.0.909
requests==2.25.1
```

So I don't know why is trying to install `airflow-0.6.tar.gz` when it's not in list of packages in my requirements file?

I've tried doing `breez stop` and `breez cleanup` which didn't help.

What would be best way to install all Python packages dependencies for all DAGs anyway?

Thanks, 
H

### What you think should happen instead

It should install Python packages and run Airflow.  

### How to reproduce

In `init.sh` put:
```
pip install --upgrade pip
pip install -r /opt/airflow/dags/requirements.txt
```

Create `requirements.txt` and put it in `dags/requirements.txt`. Add list of some Python packages that needs to be installed:
```
pyOpenSSL==22.0.0
boto3==1.26.50
botocore==1.23.34
jsonpath_ng==1.5.3
numpy==1.24.1
pandas==1.5.2
pendulum==2.1.2
psycopg2==2.9.5
psycopg2_binary==2.9.5
python_dateutil==2.8.2
redshift_connector==2.0.909
requests==2.25.1
```

You can now run:
```
breeze start-airflow --python 3.10  --load-example-dags --backend postgres --postgres-version 14 --use-airflow-version 2.5.0 │
 --airflow-constraints-reference "https://raw.githubusercontent.com/apache/airflow/constraints-2.5.0/constraints-3.10.txt" |
 --airflow-extras [pandas,amazon,postgres,sftp]
```

### Operating System

Ubuntu 22.04 on WSL2 Windows 11

### Versions of Apache Airflow Providers

pandas,amazon,postgres,sftp

### Deployment

Docker-Compose

### Deployment details

Good version of Docker: 20.10.23.
Good version of docker-compose: 2.15.1

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment, marked as answer]
This is at most troubleshooting of your issue. You shoudl check if you are really installing the requirements you think you are installing - I believe not. There is no magic in init.sh. It just executes the code at startup in the way you would run it manually. You should debug your issue by jus trying out the same commands after entering breese and looking why they fail. I guess your file in /dags folder is not the one you think it is.

**Reply by potiuk** [reply]
I think the problem the "dags" folder is not mapped to inside the container. The folders are selectively mounted you likely should put your file in "files" and access it from "/files/requirements.txt".

**Comment by hrvoje-intuita** [comment]
You were right, it worked ok from  "`/files/requirements.txt`".
DAG folder is mapped on the end properly but I guess at the time when `init.sh` was executed it hasn't been mapped yet.
